cmd/internal/encryption/encryption.go

```diff
@@ -0,0 +1,192 @@
+package encryption
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"errors"
+	"fmt"
+	"io"
+	"log/slog"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"unicode"
+)
+
+// Suffix is appended on encryption and removed on decryption from given input
+const suffix = ".aes"
+
+// Encrypter is used to encrypt/decrypt backups
+type Encrypter struct {
+	key string
+	log *slog.Logger
+}
+
+// New creates a new Encrypter with the given key.
+// The key should be 16 bytes (AES-128), 24 bytes (AES-192) or
+// 32 bytes (AES-256)
+func New(log *slog.Logger, key string) (*Encrypter, error) {
```

Comment:
Maybe we should only support one key length to have simpler code here and AES-256 is not much slower than AES-128

```
openssl speed aes
Doing aes-128-cbc for 3s on 16 size blocks: 279659384 aes-128-cbc's in 3.00s
Doing aes-128-cbc for 3s on 64 size blocks: 101742267 aes-128-cbc's in 3.00s
Doing aes-128-cbc for 3s on 256 size blocks: 26170389 aes-128-cbc's in 3.00s
Doing aes-128-cbc for 3s on 1024 size blocks: 6590398 aes-128-cbc's in 3.00s
Doing aes-128-cbc for 3s on 8192 size blocks: 823868 aes-128-cbc's in 2.99s
Doing aes-128-cbc for 3s on 16384 size blocks: 409378 aes-128-cbc's in 2.99s
Doing aes-192-cbc for 3s on 16 size blocks: 242289444 aes-192-cbc's in 3.00s
Doing aes-192-cbc for 3s on 64 size blocks: 82943387 aes-192-cbc's in 3.00s
Doing aes-192-cbc for 3s on 256 size blocks: 22219614 aes-192-cbc's in 3.00s
Doing aes-192-cbc for 3s on 1024 size blocks: 5528655 aes-192-cbc's in 3.00s
Doing aes-192-cbc for 3s on 8192 size blocks: 655650 aes-192-cbc's in 3.00s
Doing aes-192-cbc for 3s on 16384 size blocks: 328250 aes-192-cbc's in 3.00s
Doing aes-256-cbc for 3s on 16 size blocks: 205438121 aes-256-cbc's in 3.00s
Doing aes-256-cbc for 3s on 64 size blocks: 68722942 aes-256-cbc's in 3.00s
Doing aes-256-cbc for 3s on 256 size blocks: 17695372 aes-256-cbc's in 3.00s
Doing aes-256-cbc for 3s on 1024 size blocks: 4471417 aes-256-cbc's in 3.00s
Doing aes-256-cbc for 3s on 8192 size blocks: 546280 aes-256-cbc's in 3.00s
Doing aes-256-cbc for 3s on 16384 size blocks: 277612 aes-256-cbc's in 3.00s
version: 3.0.13
built on: Tue Aug 20 17:05:36 2024 UTC
options: bn(64,64)
compiler: gcc -fPIC -pthread -m64 -Wa,--noexecstack -Wall -fzero-call-used-regs=used-gpr -DOPENSSL_TLS_SECURITY_LEVEL=2 -Wa,--noexecstack -g -O2 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -ffile-prefix-map=/build/openssl-WgbqpY/openssl-3.0.13=. -fstack-protector-strong -fstack-clash-protection -Wformat -Werror=format-security -fcf-protection -fdebug-prefix-map=/build/openssl-WgbqpY/openssl-3.0.13=/usr/src/openssl-3.0.13-0ubuntu3.4 -DOPENSSL_USE_NODELETE -DL_ENDIAN -DOPENSSL_PIC -DOPENSSL_BUILDING_OPENSSL -DNDEBUG -Wdate-time -D_FORTIFY_SOURCE=3
CPUINFO: OPENSSL_ia32cap=0x7ffaf3ffffebffff:0x984007bc239c27eb
The 'numbers' are in 1000s of bytes per second processed.
type             16 bytes     64 bytes    256 bytes   1024 bytes   8192 bytes  16384 bytes
aes-128-cbc    1491516.71k  2170501.70k  2233206.53k  2249522.52k  2257233.00k  2243227.14k
aes-192-cbc    1292210.37k  1769458.92k  1896073.73k  1887114.24k  1790361.60k  1792682.67k
aes-256-cbc    1095669.98k  1466089.43k  1510005.08k  1526243.67k  1491708.59k  1516131.67k

```